pmd-ui/src/main/java/net/sourceforge/pmd/util/fxdesigner/model/XPathSuggestions.java

```diff
@@ -0,0 +1,98 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.util.fxdesigner.model;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Enumeration;
+import java.util.List;
+
+
+
+
+public class XPathSuggestions {
+    private List<String> xPathSuggestions = new ArrayList<>();
+    private String packagename;
+    private String language;
+
+    public XPathSuggestions(String language) {
+        this.packagename = "net.sourceforge.pmd.lang." + language + ".ast";
+        this.language = language;
+    }
+
+
+    public List<String> getXPathSuggestions() throws IOException, ClassNotFoundException {
+        return evaluateXpathSuggestions(createList(getClasses(packagename)));
+    }
+
+    private List<String> createList(Class[] classArray) {
+        List<Class> fileNameList = Arrays.asList(classArray);
+        List<String> foo = new ArrayList<>();
+
+        for (Class c : fileNameList) {
+            if (c.getName().contains("AST")) {
```

Comment:
Abstracting that into a function (`isAstNode(Class)`) would be better. The filtering misses some things too, for example, it should filter out classes whose names contain a `$` symbol (nested classes), and classes that are not concrete.